src/ringmesh/geomodel/geomodel_builder.cpp

```diff
@@ -1006,6 +1032,217 @@ namespace RINGMesh
         geometry.cut_surfaces_by_internal_lines();
     }
 
+    void GeoModelBuilder< 2 >::build_surfaces_from_corners_and_lines()
+    {
+        if( geomodel_.nb_surfaces() > 0 )
+        {
+            return;
+        }
+        // Each side of each Line is in one Surface(+side is first)
+        std::vector< index_t > line_2_surface;
+        std::vector< std::vector< OrientedLine > > surface_boundary_lines;
+        find_surfaces_boundary_lines( line_2_surface, surface_boundary_lines );
+
+        check_internal_floating_lines( line_2_surface,
+            static_cast< index_t >( surface_boundary_lines.size() ) );
+
+        // Generate surface polygons
+        build_surface_polygons( surface_boundary_lines );
+        find_exterior_and_remove_it( surface_boundary_lines );
+
+        // Update topology
+        set_surface_line_boundary_relationships( surface_boundary_lines );
+    }
+
+    void GeoModelBuilder< 2 >::find_surfaces_boundary_lines(
+        std::vector< index_t >& line_2_surface,
+        std::vector< std::vector< OrientedLine > >& surface_boundary_lines )
+    {
+        line_2_surface.resize( 2 * this->geomodel_.nb_lines(), NO_ID );
```

Comment:
80 lines of codes.. I think we should add a limit we should not cross. Probably something like 50. Could it be possible to refactor this method? It will reduce the probability of bug if we could understand it easily.